src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/CollectionModelBinder.cs

```diff
@@ -164,12 +161,20 @@ protected object CreateInstance(Type targetType)
                 };
 
                 object boundValue = null;
-                var result =
-                    await bindingContext.OperationBindingContext.ModelBinder.BindModelAsync(innerBindingContext);
-                if (result != null && result.IsModelSet)
+
+                using (bindingContext.PushContext(
+                    elementMetadata,
+                    fieldName: bindingContext.FieldName,
+                    modelName: bindingContext.ModelName,
+                    model: null))
```

Comment:
Why isn't the loop over `values` inside this `using`? Should only need to reset `Result` to `null` inside the loop.